src/java.base/share/classes/java/util/zip/DeflaterInputStream.java

```diff
@@ -75,6 +75,9 @@ private void ensureOpen() throws IOException {
      * @throws NullPointerException if {@code in} is null
      */
     public DeflaterInputStream(InputStream in) {
+        // "in" being null isn't allowed. we use a null check for "in"
+        // merely to avoid an unnecessary instance creation of the Deflater
+        // for such erroneous cases.
```

Comment:
I see this same comment has been added in 3 places but it's not easy to read and I don't think is needed.

Reply:
I've now updated the PR to remove this code comment.